lib/Fmt_ast.ml

```diff
@@ -106,18 +106,15 @@ let fmt_elements_collection ?(first_sep = true) ?(last_sep = true)
   in
   list_fl xs fmt_one
 
-let fmt_expressions c width sub_exp exprs fmt_expr p =
+let box_collec c is_simple elements =
   match c.conf.break_collection_expressions with
-  | `Fit_or_vertical -> fmt_elements_collection p fmt_expr exprs
-  | `Wrap ->
-      let is_simple x = is_simple c.conf width (sub_exp x) in
-      let break x1 x2 = not (is_simple x1 && is_simple x2) in
-      let grps = List.group exprs ~break in
-      let fmt_grp ~first:first_grp ~last:last_grp exprs =
-        fmt_elements_collection ~first_sep:first_grp ~last_sep:last_grp p
-          fmt_expr exprs
-      in
-      list_fl grps fmt_grp
```

Comment:
This code is not maintained in the PR (even as "wrap" is maintained) because it doesn't fit the new architecture and this behavior is improved by "smart"

Reply:
This option is deprecated right ?
I couldn't find an example of regression with test_branch, what did I do wrong ?

Reply:
> This option is deprecated right ?

Yes, `break-collection-expressions = wrap` would then disappear and only `fit-or-vertical` will remain. This PR is an alternative that merges the 2 behaviors. Maybe it's better to effectively remove the option in `main` before changing the behavior in this PR.

> I couldn't find an example of regression with test_branch, what did I do wrong ?

I will have a look

Reply:
I didn't have any diff as well, but I plan to remove the two existing values to replace them by the new ones, that would make more sense I think.